core/trie_test.go

```diff
@@ -207,17 +166,21 @@ func TestTriePut(t *testing.T) {
 			root:  nil,
 		},
 	}
-	for idx, test := range tests {
-		if err := trie.Put(test.key, test.value); err != nil {
-			t.Errorf("TestTriePut: Put() failed at test #%d", idx)
-		}
-		if value, err := trie.Get(test.key); err != nil || !value.Equal(test.value) {
-			t.Errorf("TestTriePut: Get() failed at test #%d", idx)
-		}
-		if test.root != nil && !test.root.Equal(trie.root) {
-			t.Errorf("TestTriePut: Unexpected root at test #%d", idx)
+	DoOnTempTrie(251, func(trie *Trie) error {
+		for idx, test := range tests {
+			if err := trie.Put(test.key, test.value); err != nil {
+				t.Errorf("TestTriePut: Put() failed at test #%d", idx)
+			}
+			if value, err := trie.Get(test.key); err != nil || !value.Equal(test.value) {
+				t.Errorf("TestTriePut: Get() failed at test #%d", idx)
+			}
+			if test.root != nil && !test.root.Equal(trie.root) {
+				t.Errorf("TestTriePut: Unexpected root at test #%d", idx)
+			}
 		}
-	}
+
+		return nil
+	})
```

Comment:
Can your please explain why the `trie.Get` works when `txn.Commit()` is yet to be called? Is it because the Get calls are being made on the same database Transactions on which Set was called?

Reply:
A transaction can see the changes it made to the DB without calling `Commit()`. After `Commit()`, only newly created transactions can see those changes.

So;
```

db := newDb

tx1 := db.newTransaction
tx2 := db.newTransaction

tx1.Put(key, value)
assert(value == tx1.Get(key)) // pass

tx2.Get(key) // not found

tx1.Commit()

tx2.Get(key) // not found

tx3 := db.newTransaction
assert(value == tx3.Get(key)) // pass
```

Reply:
Thank you for making that clear!